pilot/pkg/networking/plugin/mixer/mixer_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixer
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	mccpb "istio.io/api/mixer/v1/config/client"
+	"istio.io/istio/pilot/pkg/model"
+	context "istio.io/istio/pilot/pkg/model"
+)
+
+func TestTransportConfig(t *testing.T) {
```

Comment:
I understand wanting to move fast and have good code coverage - but for this kind of change an e2e test would be far more appropriate. Is the annotation really propagated from pod annotation all the way to the mixer, and the right config generated ? 

I would settle for just an example config, with the annotations set, in the tools/ repo
where we do the real testing - so at least it is easier to manually test this until 
it is automated, and we can get some meaningful metrics. 

Maybe - this test only checks that if you set a value in a hashmap - you get it back.

Reply:
This is similar to pod policy annotation. I manually verified that propagation works. I think @howardjohn also needs to manually verify this e2e. It's not entirely obvious how to simulate and validate retries. Perhaps, killing policy and monitoring the proxy counters?